### PR TITLE
Added support for a Vue file and positioning options

### PR DESCRIPTION
**Pull request by calebsmithdev**
This PR includes support for the new `dimAmount` for NS-Vue users, as well as a dedicated Vue template to continue to build on for future additions. 

The `ModalStack` file has been modified to allow for horizontal and vertical positioning per the `verticalPosition` and `horizontalPosition` parameters. They use the same options as `verticalAlignment` and `horizontalAlignment`.

The documentation has been updated to show the new parameters as well as how to use for NS Vue users.

This is also only my second community PR, so please let me know if I can improve.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,42 +23,85 @@ Call the `overrideModalViewMethod()` once before starting the app and register t
...
-    dimAmount: 0.5 // Sets the alpha of the background dim
-} as ExtendedShowModalOptions);
+mainPage.showModal('./modal', {
+  context: "I'm the context",
+  closeCallback: (response: string) => console.log('Modal response: ' + response),
+  dimAmount: 0.5 // Sets the alpha of the background dim
+} as ExtendedShowModalOptions)
+```
+
+You can pass extended options like this:
```
Comment (mukaschultze) [inline README.md]: This is duplicated

File: README.md [hunk 2]
```diff
@@ -80,9 +123,7 @@ Wrap your modal component with a `ModalStack` tag to layout the elements in a co
 
 ```html
 <ModalStack dismissEnabled="true" class="modal-container">
-    <StackLayout class="modal">
-        <Label text="Hi, I'm your modal" class="text-center" textWrap="true"></Label>
-    </StackLayout>
+  <StackLayout class="modal"> <label text="Hi, I'm your modal" class="text-center" textWrap="true"></label> </StackLayout>
```
Comment (mukaschultze) [inline README.md]: `Label` is case sensitive, and it's best to keep it on a new line to prevent scroll bars

File: src/modal-stack.ts [hunk 3]
```diff
@@ -1,60 +1,61 @@
...
 
-@CSSType("ModalStack")
+@CSSType('ModalStack')
 export class ModalStack extends StackLayout {
+  dismissEnabled: string = 'true';
+  verticalPosition: VerticalAlignment = 'middle';
+  horizontalPosition: HorizontalAlignment = 'center';
 
-    dismissEnabled: string = "true";
+  constructor() {
```
Comment (mukaschultze) [inline src/modal-stack.ts]: Keep the 4 spaces indent, please

**Review by mukaschultze — CHANGES_REQUESTED** [review]
I know literally nothing of vue, but the code seems good, there are just a few notes that I think are worth fixing

**Comment by calebsmithdev** [comment]
Thanks for the feedback! The README issues were interesting.. I guess my autoformat on save caught the label. I adjusted the two pages that I edited to 4 spaces and pushed them up. Overall - I kept the Vue code almost identical to the NS Vue package so it could be easy to update if it ever changes.

**Review by mukaschultze — APPROVED** [review]

**Comment by mukaschultze** [comment]
Thanks for contributing

**Comment by calebsmithdev** [comment]
Absolutely! It's a pleasure to have such an awesome package in my apps.

**Pull request merged**